### PR TITLE
restructured intro

### PR DESCRIPTION
**Pull request by vomov**

### CODE REVIEW

File: intro.md [hunk 1]
```diff
@@ -10,75 +10,89 @@ title: "Introduction to DSH"
 });-->
 
 <!-- .slide: data-background="./images/kpn-intro-bg-md.jpg" -->
+
 # An introduction to the Data Services Hub
 
+Note: audience should have had a generic talk by bisdev before this part
```
Comment (mtjon) [inline intro.md]: Typo bisdev -> bizdev
Why is this note here? Do you want the trainer to validate this pre-requisite during the introduction?

File: intro.md [hunk 2]
```diff
@@ -10,75 +10,89 @@ title: "Introduction to DSH"
...
+
 <!--s-->
-# What is the DSH?
+## What is the DSH?
+
+The DSH is an
+
 AWESOME <!-- .element: class="fragment" data-fragment-index="1" -->
 
+Streaming Data Platform <!-- .element: class="fragment" data-fragment-index="2" -->
```
Comment (mtjon) [inline intro.md:23]: You intend the `AWESOME` to appear first and have `Streaming Data Platform` appear upon another click? Seems undesirable to me. Or are `AWESOME` and `Streaming Data Platform` two list items? In which case I'd expect something akin to:
```
The DSH is:
* AWESOME
* a Streaming Data Platform
```

File: intro.md [hunk 3]
```diff
@@ -10,75 +10,89 @@ title: "Introduction to DSH"
...
-- Data platform <!-- .element: class="fragment" data-fragment-index="1" -->
-- Data (as events) <!-- .element: class="fragment" data-fragment-index="2" -->
-- Sharing <!-- .element: class="fragment" data-fragment-index="3" -->
-- Processing <!-- .element: class="fragment" data-fragment-index="4" -->
-- Scalable <!-- .element: class="fragment" data-fragment-index="5" -->
-- Secure <!-- .element: class="fragment" data-fragment-index="6" -->
-- Low-latency <!-- .element: class="fragment" data-fragment-index="7" -->
 
-Note: incomplete list, obvs'.
+## Definition: platform
```
Comment (mtjon) [inline intro.md:28]: So the idea is that we no longer list the `Why is the DSH Awesome` and instead start talking about definitions?

File: intro.md [hunk 4]
```diff
@@ -10,75 +10,89 @@ title: "Introduction to DSH"
...
-# What is the DSH?
+Key concepts:
+- Data as low-latency events (streams) <!-- .element: class="fragment"-->
+- Real-time processing <!-- .element: class="fragment"-->
+- Data sharing <!-- .element: class="fragment"-->
+- Scalable platform <!-- .element: class="fragment"-->
+- Secure <!-- .element: class="fragment"-->
 
-A platform that does something awesome with <!-- .element: class="fragment" data-fragment-index="1" --> _streaming data_ <!-- .element: class="fragment" data-fragment-index="1" -->
+What does the DSH do better than others?  <!-- .element: class="fragment"-->
```
Comment (mtjon) [inline intro.md:48]: I don't quite get what you're trying to communicate here or for what reason you changed it from the old list. What did you try to improve?

File: intro.md [hunk 5]
```diff
@@ -10,75 +10,89 @@ title: "Introduction to DSH"
...
 https://aws.amazon.com/streaming-data
 
+A streaming platform should be able to handle thousands of data sources  <!-- .element: class="fragment"-->
+
 Note: this is a quote, and as such only applies to the view of the person making the quote
 
 <!--v-->
 
-## Our definition: Streaming Data
+## We expect more
```
Comment (mtjon) [inline intro.md:70]: Why block-quote this list?

File: intro.md [hunk 6]
```diff
@@ -10,75 +10,89 @@ title: "Introduction to DSH"
...
+> A streaming data platform should be able to:
+> - handle hundreds of thousands of sources
+> - send data to hundreds of thousands of sinks
+> - process (clean, refine, aggregate, combine) data
+> - share data streams with other parties
+> - do all of this, with __high security standards__
 
-> A streaming data platform should also be able to continuously send selected
-> data records to thousands of data sinks.
+Note: security is the biggest part of the DSH, and will annoy you
```
Comment (mtjon) [inline intro.md]: Not sure about the phrasing. Is it the *biggest* part? It shouldn't annoy a user if they know *why* it's implemented in a certain way. Which is what we are for.

File: intro.md [hunk 7]
```diff
@@ -10,75 +10,89 @@ title: "Introduction to DSH"
...
 
 <!--v-->
-# Data Streams
+## Data Streams
```
Comment (mtjon) [inline intro.md]: So this is still part of `Why is the DSH awesome`? Again, I'd like to hear the reasoning of the restructuring you apply.

File: intro.md [hunk 8]
```diff
@@ -93,28 +107,61 @@ $$
...
+Note: MQTT and Kafka are equally useful, but for very different reasons
+
 <!--s-->
 ## MQTT
-- Messaging protocol
-- ISO/IEC 20922 and, OASIS standard
-- Lightweight messaging protocol
-- Suitable for many simultaneous connections
-- Widespread in the *Internet of Things*
+- _Lightweight_ messaging protocol
```
Comment (mtjon) [inline intro.md]: Combine with next bullet: `_(Extremely, in the words of MQTT.org) Lightweight_ pub/sub messaging transport`

File: intro.md [hunk 9]
```diff
@@ -93,28 +107,61 @@ $$
...
 <!--s-->
 ## MQTT
-- Messaging protocol
-- ISO/IEC 20922 and, OASIS standard
-- Lightweight messaging protocol
-- Suitable for many simultaneous connections
-- Widespread in the *Internet of Things*
+- _Lightweight_ messaging protocol 
+- Allows subscribe and publish <!-- .element: class="fragment" data-fragment-index="1"-->
+- Widespread in the  <!-- .element: class="fragment" data-fragment-index="2" --> *Internet of Things* <!-- .element: class="fragment" data-fragment-index="2" -->
```
Comment (mtjon) [inline intro.md]: `Widespread use` seems (more) correct?

File: intro.md [hunk 10]
```diff
@@ -93,28 +107,61 @@ $$
...
+- Suitable for many simultaneous connections <!-- .element: class="fragment" data-fragment-index="3" -->
+- ISO/IEC 20922 and OASIS standard <!-- .element: class="fragment" data-fragment-index="4" -->
+- Subscription-based with fine-grained access control<!-- .element: class="fragment" -->
```
Comment (mtjon) [inline intro.md]: No need for `Subscription-based with` if we already mention it is a pub/sub transport

File: intro.md [hunk 11]
```diff
@@ -93,28 +107,61 @@ $$
...
+
+Note: MQTT is a very good way to get data on the DSH. As a sidenote:
```
Comment (mtjon) [inline intro.md:116]: Repo is in English, notes should be too.

File: intro.md [hunk 12]
```diff
@@ -93,28 +107,61 @@ $$
...
+<!--v-->
+## Do I have to use MQTT?
+
+- Maybe you don't need the features MQTT brings? <!-- .element: class="fragment" -->
+- Maybe MQTT is hard to implement in your case?  <!-- .element: class="fragment" -->
+- Maybe another protocol is already implemented?  <!-- .element: class="fragment" -->
+
+We allow tenants to write custom protocol adapters  <!-- .element: class="fragment" -->
+ 
+Note: we do not say other adapters are in the works. Kafka is the backbone, MQTT is one very cool way of ingesting/extruding data
```
Comment (mtjon) [inline intro.md]: Basically a user may either be 1.) unable or 2.) unwilling to use MQTT. The message is that tenants are free to bring/ build their own protocol adapters. I'd argue the three examples given are all of the category 'unwilling', perhaps change one out for an example where the user is unable to change the source/sink and therefore needs to make use of the available transport other than MQTT.

File: intro.md [hunk 13]
```diff
@@ -93,28 +107,61 @@ $$
...
+    - PayPal
+    - Spotify
+    - Salesforce
+    - Uber
+    - Yelp
+    - Pinterest
+    - eBay
+    - Cloudflare
+
+Note: remove items based on customer
```
Comment (mtjon) [inline intro.md]: Don't want to edit slidedeck before every use. Suggestion: keep a few genergic big-name examples. Move rest to notes and mention them verbally, based on audience.

File: intro.md [hunk 14]
```diff
@@ -128,14 +175,16 @@ Note: more about MQTT later
   - _must_ have few sources/sinks
   - sources/sinks reside inside DSH
 
-$$ \text{MQTT} \cdot \frac{sources}{sinks} \approx \text{Kafka} \cdot \frac{sources}{sinks} $$ <!-- .element: class="fragment" data-fragment-index="1" -->
+<!-- $$ \text{MQTT} \cdot \frac{sources}{sinks} \approx \text{Kafka} \cdot \frac{sources}{sinks} $$ .element: class="fragment" data-fragment-index="1" -->
 
-Note: kafka sources/ sinks can also reside outside of DSH
+Note: 
+- kafka sources/sinks can also reside outside of DSH
+- Explicityly mention we're going to make the thing visual now
```
Comment (mtjon) [inline intro.md]: typo: `Explicityly` -> `explicitly`
also: 'the thing'?

File: intro.md [hunk 15]
```diff
@@ -144,87 +193,70 @@ Note: kafka sources/ sinks can also reside outside of DSH
...
 ![dsh-overview-2](images/dsh/dsh-kafkamqtt.svg)<!-- .element: class="stretch" style="background:none; border:none; box-shadow:none;" width="100%" -->
 
+Note: the core of the DSH messaging is a Kafka bus. Note the MQTT bridge.
+
 <!--v-->
 <!-- .slide: data-transition="fade" -->
 ## Overview
-
 ![dsh-overview-2](images/dsh/dsh-sourcessinks.svg)<!-- .element: class="stretch" style="background:none; border:none; box-shadow:none;" width="100%" -->
+Note: we connect sources and sinks through MQTT bridges. The messages they send, end up on a 'stream'.
```
Comment (mtjon) [inline intro.md:196]: We may want to call them `topics`. We have three types of topics; `stream`, `internal`, `scratch`. It is correct that MQTT messages end up on a topic of the `stream`-variant, though.

File: intro.md [hunk 16]
```diff
@@ -144,87 +193,70 @@ Note: kafka sources/ sinks can also reside outside of DSH
...
- /platform/stream/topic/#
- ```
- <!-- .element: class="fragment" data-fragment-index="2" -->
-
-Note: wildcard subscriptions later, in the MQTT part
+<!-- .slide: data-transition="fade" -->
+## Overview
+![dsh-overview-2](images/dsh/dsh-sourcessinks-singlestream.svg)<!-- .element: class="stretch" style="background:none; border:none; box-shadow:none;" width="100%" -->
+Note: access to this stream has two sides; from outside, and from inside (the DSH). First, we'll deal with the 'inside'. 
+Let's say the data source (sensor) is _point at member of audience_ yours. That makes the data, and the stream, yours as well.
```
Comment (mtjon) [inline intro.md:214]: These 'presentation instructions' are not going to be used during a training and are awkward to read post-presentation as an attendee. Suggest removing.

File: intro.md [hunk 17]
```diff
@@ -144,87 +193,70 @@ Note: kafka sources/ sinks can also reside outside of DSH
...
-
-<!--s-->
 <!-- .slide: data-transition="fade" -->
-
-# Kafka
-
-![dsh-overview-2](images/dsh/dsh-kafkamqtt.svg)<!-- .element: class="stretch" style="background:none; border:none; box-shadow:none;" width="100%" -->
+## Overview
+![dsh-overview-2](images/dsh/dsh-sourcessinks-singletenant.svg)<!-- .element: class="stretch" style="background:none; border:none; box-shadow:none;" width="100%" -->
+Note: _point at member of audience_ this is your tenant
```
Comment (mtjon) [inline intro.md:220]: See earlier comment about this type of note

File: intro.md [hunk 18]
```diff
@@ -144,87 +193,70 @@ Note: kafka sources/ sinks can also reside outside of DSH
...
-
-## Kafka
-Three Kafka stream-types
--  _stream._ topic
--  _internal._ topic 
--  _scratch._ topic
+<!-- .slide: data-transition="fade" -->
+## Overview
+![dsh-overview-2](images/dsh/dsh-sourcessinks-singletenant-stream.svg)<!-- .element: class="stretch" style="background:none; border:none; box-shadow:none;" width="100%" -->
+Note: within your tenant, you run your stuff on your stream.
```
Comment (mtjon) [inline intro.md]: What do you mean by this? How are topics "within your tenant"? How do you _run 'stuff' on_ your stream?

File: intro.md [hunk 19]
```diff
@@ -144,87 +193,70 @@ Note: kafka sources/ sinks can also reside outside of DSH
...
 
 <!--v-->
 <!-- .slide: data-transition="fade" -->
-![kafkastreams](images/dsh/dsh-kafkastreams.svg)<!-- .element: class="stretch" style="background:none; border:none; box-shadow:none;" width="100%" -->
+## Overview
+![dsh-overview-2](images/dsh/dsh-sourcessinks-scratch.svg)<!-- .element: class="stretch" style="background:none; border:none; box-shadow:none;" width="100%" -->
+Note: if you have multiple things running in your tenant, you can make use of a stream type that is not available outside your tenant. You cannot share this stream.
```
Comment (mtjon) [inline intro.md]: Recurring theme; please avoid 'stuff', 'things', etc. In this case you'd be running multiple _services_ in your tenant.
You make use of a 'topic' type, not a 'stream' type. A 'stream' is a type of 'topic'.

File: intro.md [hunk 20]
```diff
@@ -247,9 +279,9 @@ Note: example ECG sensors. First, all data is processed at the computer of the d
 <!--v-->
 ## Many ways to process the data
 
-- Many frameworks for stream processing
+- Many frameworks for (stream) processing
```
Comment (mtjon) [inline intro.md:276]: why put parenthesis here? The topic is stream processing, after all.

File: intro.md [hunk 21]
```diff
@@ -247,9 +279,9 @@ Note: example ECG sensors. First, all data is processed at the computer of the d
...
 - No framework fits all use-cases
-- DSH does not dictate a framework
+- DSH does not dictate a framework; you can use whatever you want
```
Comment (mtjon) [inline intro.md]: I'd leave the bullets as they were.

File: intro.md [hunk 22]
```diff
@@ -258,45 +290,43 @@ Note: analogy: you can bring your project to Black&Decker, and then you'll have
 <!--s-->
 # Security nightmare
 
-- Need to allow other people on your platform for proximity
-- And they can use whatever software they want on the platform
+- You share the platform with others <!-- .element: class="fragment"-->
+- You can use whatever they want <!-- .element: class="fragment"-->
+- Streams have access control lists <!-- .element: class="fragment"-->
+- Tenants are separated <!-- .element: class="fragment"-->
```
Comment (mtjon) [inline intro.md]: What was the idea of your changes? Before, the bullets give two required platform features that the trainer can use to build an urgency for proper management of access control. The first two new bullets have the same 'outcome', but without any reasoning why it was necessary. The last two new bullets are mitigating the 'nightmare' which seems like they don't belong in the same list.

File: intro.md [hunk 23]
```diff
@@ -258,45 +290,43 @@ Note: analogy: you can bring your project to Black&Decker, and then you'll have
...
 
 <!--v-->
 <!-- .slide: data-transition="fade" -->
 ## Base DSH
 ![DC/OS](images/dsh/dsh-tenants.svg)<!-- .element: class="stretch" style="background:none; border:none; box-shadow:none;" width="100%" -->
+Note: in your tenant, you run docker containers. Correct use of docker is _forced_. Docker is already quite secure, but the DSH ensures you can't hack your way out of your container (you can't be root)
```
Comment (mtjon) [inline intro.md]: I'd phrase this differently.
Docker containers are meant to isolate processes from the host's environment they run on. This is extremely important when sharing underlying hosts with other processes, possibly owned by other tenants. Therefore, we regulate the use of Docker&mdash;e.g. your process mus run with a specific uid.

File: prerequisites.md [hunk 24]
```diff
@@ -1,8 +1,9 @@
 # Prerequisites to the Tenant-in-a-day-training
 ## General information
 
-This guide is ensure the training proceeds as desired, with all participants having fullfilled all basic requirements. 
+This guide is ensure the training proceeds as desired, with all participants having fullfilled all basic requirements.
```
Comment (mtjon) [inline prerequisites.md]: Typo; "This guide is ensure" -> 'This guide aims to ensure'?
Suggestion; "requirements" -> 'pre-requisites'

**Review by mtjon — CHANGES_REQUESTED** [review]
* Please remove `notes.md` and `preparation.md` from the PR.
* Not sure about the changes in `prerequisistes.md` what did they resolve?
* Next time; be specific of what the PR should resolve, ideally referring to an existing issue. it is called 'restructured intro' now, but multiple other files got edited too.

File: intro.md (hunk 5 above)
Comment (vomov) [reply intro.md:70]: Intended to be a quote from us, but it looks odd. Fix'd to a normal list.

File: intro.md (hunk 20 above)
Comment (vomov) [reply intro.md:276]: That's to deliberately start a discussion, which apparently works, thanks for testing 🤣 
Not all analyses are stream-based, some are batch-based or event-based. To ensure the trainees don't experience any potential limitations, I want to give space for a brief discussion in the direction of use-cases.

File: intro.md (hunk 4 above)
Comment (vomov) [reply intro.md:48]: Reduced the amount of text on the screen, to draw attention away from the presentation and towards the presenter (who is more interesting).

File: intro.md (hunk 3 above)
Comment (vomov) [reply intro.md:28]: Somewhat; the flow used to be "this is the DSH", with definitions appearing either directly after they were relevant, or not at all. I've moved some to the notes, but some definitions are made explicit and moved forward.

File: intro.md (hunk 1 above)
Comment (vomov) [reply intro.md]: Yes, but should be more specific and should be somewhere else. Fixed.

File: intro.md (hunk 2 above)
Comment (vomov) [reply intro.md:23]: That's to emphasize the `awesome` part, which works better when spoken. They are somewhat separate, as the DSH is a streaming data platform with a boatload of if's, but's, and however's.

File: intro.md (hunk 6 above)
Comment (vomov) [reply intro.md]: Good point, fix'd.

File: intro.md (hunk 7 above)
Comment (vomov) [reply intro.md]: This is one of the definitions, but the first part is no longer needed. Fix'd.

File: intro.md (hunk 8 above)
Comment (vomov) [reply intro.md]: It should be two bullets, but the `messaging protocol`-part makes it confusing. Switched things around, which also emphasizes the `lightweight`-part.

File: intro.md (hunk 9 above)
Comment (vomov) [reply intro.md]: Better, fix'd.

File: intro.md (hunk 10 above)
Comment (vomov) [reply intro.md]: Correct, fix'd.

File: intro.md (hunk 11 above)
Comment (vomov) [reply intro.md:116]: Good point, apologies, fix'd.

File: intro.md (hunk 12 above)
Comment (vomov) [reply intro.md]: These three are to better connect with the issues the customer may think exist when implementing MQTT, while at the same time retaining the option of implementing a custom protocol adapter.
"don't need the features" suggests it has features you *may* need
"hard to implement" suggests implementation can be hard, but can also be easy
"another protocol implemented" barely contains the word 'inferior' before "protocol", but will make customers think. And this is the bridge to the custom protocol adapters. I'm not extremely happy with how this works, but thought we'd test it during the next TioD.

File: intro.md (hunk 13 above)
Comment (vomov) [reply intro.md]: Good point, fix'd.

File: intro.md (hunk 14 above)
Comment (vomov) [reply intro.md]: Fix'd

File: intro.md (hunk 15 above)
Comment (vomov) [reply intro.md:196]: MQTT uses the term `topic` in a different way, and common usage of the term is even different. I think I'd like to alter some of the terminology and add an appendix later, but that is beyond the scope of this commit. Shall we keep it like this, for now?

File: intro.md (hunk 16 above)
Comment (vomov) [reply intro.md:214]: I disagree: the instructions are for us, and appear during the presentation on the speaker view. It is unlikely the attendee sees them post-training, but if he does, it shouldn't be an issue.

File: intro.md (hunk 18 above)
Comment (vomov) [reply intro.md]: Agreed, this is not clear. Fixed.

File: intro.md (hunk 19 above)
Comment (vomov) [reply intro.md]: Agreed on the stuff (ha), and fix'd. Terminology is still an issue, which I'd like to revisit (and then put in another commit).

File: intro.md (hunk 21 above)
Comment (vomov) [reply intro.md]: Fix'd.

File: intro.md (hunk 22 above)
Comment (vomov) [reply intro.md]: Good point, removed the last two bullets.
The first two bullets attempt to make the sharing and freedom explicit. Should they change?

File: intro.md (hunk 23 above)
Comment (vomov) [reply intro.md]: Yeah, that's better. Altered the text.

File: prerequisites.md (hunk 24 above)
Comment (vomov) [reply prerequisites.md]: Fix'd.

File: intro.md (hunk 2 above)
Comment (mtjon) [reply intro.md:23]: Fine, we'll see how it works.

File: intro.md [hunk 25]
```diff
@@ -93,28 +102,60 @@ $$
...
+<!--v-->
+## Do I have to use MQTT?
+
+- Maybe you don't need the features MQTT brings? <!-- .element: class="fragment" -->
+- Maybe MQTT is hard to implement in your case?  <!-- .element: class="fragment" -->
+- Maybe another protocol is already implemented?  <!-- .element: class="fragment" -->
+
+We allow tenants to write custom protocol adapters  <!-- .element: class="fragment" -->
+
+Note: we do not say other adapters are in the works. Kafka is the backbone, MQTT is one very cool way of ingesting/extruding data
```
Comment (mtjon) [inline intro.md:133]: Suggest removing this note. Notes are publicly visible.

File: .gitignore [hunk 26]
```diff
@@ -1,2 +1,3 @@
 export/**
 !**/.gitkeep
+.idea
```
Comment (mtjon) [inline .gitignore:3]: Are you sure `.idea` is supposed to be in the `.gitignore`-file as opposed to a contributor's individual `.git/info/exclude`?

**Review by mtjon — APPROVED** [review]

**Pull request merged**